### PR TITLE
Handle CBMC error messages

### LINKED ISSUE #1107: CBMC parser crashes when --object-bits is too small
I tried this code:

```rust
#[kani::proof]
fn main() {
    let arr: [i32; 100] = kani::any();
    assert_eq!(arr[0], arr[99]);
}
```

using the following command line invocation:

```
kani ob.rs --object-bits 5 --unwind 30 --harness main
```

with Kani version: f822a944eb1

I expected to see this happen: Verification failed, increase unwinding.

Instead, this happened:
```
$ kani ob.rs --object-bits 5 --unwind 30 --harness main 
warning: Found the following unsupported constructs:
             - 'arith_offset' intrinsic (1)
         
         Verification will fail if one or more of these constructs is reachable.
         See https://model-checking.github.io/kani/rust-feature-support.html for more details.

warning: 1 warning emitted

Checking harness main...
Traceback (most recent call last):
  File "/home/ubuntu/git/kani/scripts/cbmc_json_parser.py", line 783, in <module>
    main(sys.argv)
  File "/home/ubuntu/git/kani/scripts/cbmc_json_parser.py", line 95, in main
    return_code = transform_cbmc_output(sample_json_file_parsing,
  File "/home/ubuntu/git/kani/scripts/cbmc_json_parser.py", line 183, in transform_cbmc_output
    properties, messages = postprocess_results(properties, extra_ptr_check)
  File "/home/ubuntu/git/kani/scripts/cbmc_json_parser.py", line 287, in postprocess_results
    has_reachable_unsupported_constructs = has_check_failure(properties, GlobalMessages.UNSUPPORTED_CONSTRUCT_DESC)
  File "/home/ubuntu/git/kani/scripts/cbmc_json_parser.py", line 325, in has_check_failure
    for property in properties:
TypeError: 'NoneType' object is not iterable
```
Passing `--output-format old` shows the actual error message from CBMC:
```
$ kani ob.rs --object-bits 5 --unwind 30 --harness main --output-format old
<snip>
Runtime Symex: 0.301955s
size of program expression: 12210 steps
simple slicing removed 5 assignments
Generated 1418 VCC(s), 355 remaining after simplification
Runtime Postprocess Equation: 0.00150717s
Passing problem to propositional reduction
converting SSA
too many addressed objects: maximum number of objects is set to 2^n=32 (with n=5); use the `--object-bits n` option to increase the maximum number
```

### PR DESCRIPTION
**Pull request by zhassan-aws**
### Description of changes: 

Currently, when CBMC's output contains error messages, the CBMC json parser crashes. This PR fixes the issue through checking if there are error messages in the output, and if so, it displays them and exits without proceeding with the rest of the post-processing steps.

### Resolved issues:

Resolves #1107 

### Testing:

* How is this change tested? Added one test

* Is this a refactor change? No

### Checklist
- [X] Each commit message has a non-empty body, explaining why the change was made
- [X] Methods or procedures are documented
- [X] Regression or unit tests are included, or existing tests cover the modified code
- [X] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: tests/expected/object-bits/insufficient/main.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Checks for error message with an --object-bits value that is too small
+
+// kani-flags: --object-bits 5 --default-unwind 30
```
Comment (tedinski) [inline tests/expected/object-bits/insufficient/main.rs:6]: enable-unstable?

**Review by tedinski — APPROVED** [review]

File: tests/expected/object-bits/insufficient/main.rs (hunk 1 above)
Comment (celinval) [reply tests/expected/object-bits/insufficient/main.rs:6]: Can you use --cbmc-args instead? I think I'm just going to nuke --object-bits.

Comment (zhassan-aws) [reply tests/expected/object-bits/insufficient/main.rs:6]: Yes, will push this change once you merge #1109.

Comment (celinval) [reply tests/expected/object-bits/insufficient/main.rs:6]: Thanks

Comment (zhassan-aws) [reply tests/expected/object-bits/insufficient/main.rs]: Done.

**Review by celinval — APPROVED** [review]
It looks good to me. Do you know if a similar message will be emitted if the value given is too big and CBMC fails to allocate memory of a certain size?

**Comment by zhassan-aws** [comment]
> Do you know if a similar message will be emitted if the value given is too big and CBMC fails to allocate memory of a certain size?

Doesn't look like (or at least I couldn't reproduce it even with `--object-bits 63`).

**Pull request merged**